reduce_confusion_matrix.py

```diff
@@ -0,0 +1,116 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+# ***************************************************************************80
+#
+# reduce_confusion_matrix.py -f <input_filename> -c <no of top mispredictions>
+#
+# *****************************************************************************
+
+# standard imports
+import heapq
+
+# third party imports
+import pandas
+import click
+from sklearn.metrics import confusion_matrix
+
+# Typical phrases.csv from HF contains
+# Labelled Phrase,Detected Phrase,Intent Id,Intent Name,Top Match Intent Id,Top Match Intent Name,Top Match Score,Entropy,Uncertainty,Margin Score,Result Type
+
+@click.command()
+@click.option('-m', '--top_mispredictions', type=int, default=5, help='Number of top mispredictions')
+@click.option('-f', '--phrases_filename', type=str, default='./data/phrases.csv', help='Phrases')
+def main(phrases_filename: str, top_mispredictions: int) -> None:
+    '''Main function'''
+
+    reduce_confusion_matrix(phrases_filename, top_mispredictions)
+
+def reduce_confusion_matrix(phrases_filename: str, top_mispredictions: int) -> None:
```

Comment:
suggest have this accept pandas.DataFrame and read the file outside of the function for reusabilty.